### PR TITLE
[DC-1640] Listing Queries In `generate_site_mappings_and_ext_tables` Deletes Extension Tables

### PR DESCRIPTION
**Pull request by dcampbell-vumc**
* Converted `generate_ext_tables.py` into cleaning rule
* Removed dependency on `bq_utils.create_tables` to generate extension tables to prevent extension table data loss when listing queries
* Updated `generate_ext_tables_test.py` to reflect changes made in `generate_ext_tables.py` module
* Slight update to how queries are generated in `generate_site_mappings_and_ext_tables.py` module by `generate_ext_tables.py`
* Slight update to mock statement in `generate_site_mappings_and_ext_tables_test.py`
* Moved both `generate_ext_tables.py` and `generate_ext_tables_test.py` to respective cleaning rule folder sections

### CODE REVIEW

File: data_steward/tools/add_hpo.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Add a new HPO site to config file and BigQuery lookup tables and updates the `pipeline_table_sandbox.site_masking`
     sandbox table created in `generate_site_mappings_and_ext_tables` with any missing hpo_sites in
```
Comment (jp3477) [inline data_steward/tools/add_hpo.py:3]: Can you update the file description to not mention `generate_site_mappings_and_ext_tables`?

**Review by jp3477 — CHANGES_REQUESTED** [review]
Everything here looks good. Should an integration test for `generate_ext_tables.py` be created as well? I also added one minor suggestion.

**Comment by lrwb-aou** [comment]
@jp3477 added an integration test and it uncovered a bug.  glad you asked for this.

**Comment by lrwb-aou** [comment]
@jp3477 finally got the conflicts fixed and the tests passing.

**Comment by lrwb-aou** [comment]
If @jp3477 approves, please merge it.

**Review by jp3477 — APPROVED** [review]

**Pull request merged**